ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -447,12 +455,22 @@ boolean addBlobStore(ReplicaId replica) {
         BlobStore store = new BlobStore(replica, storeConfig, scheduler, longLivedTaskScheduler, this, diskIOScheduler,
             diskSpaceAllocator, storeMainMetrics, storeUnderCompactionMetrics, keyFactory, recovery, hardDelete,
             replicaStatusDelegates, time, accountService, null, indexPersistScheduler);
-        store.start();
+        // For Filecopy, we do not init the BlobStore since new log/index isn't required, the files will be directly
+        // copied from remote node.
+        if (!storeConfig.enableFileCopyForBootstrap) {
+          store.start();
+        }
+
         // collect store segment requirements and add into DiskSpaceAllocator
         List<DiskSpaceRequirements> storeRequirements = Collections.singletonList(store.getDiskSpaceRequirements());
```

Comment:
If `store.start()` hasn't been invoked, in the normal Replication protocol, would `store.getDiskSpaceRequirements()` work as expected?